src/games/Roms.cpp

```diff
@@ -229,14 +233,23 @@ static const RomSettings* roms[] = {
     new ZaxxonSettings(),
 };
 
-/* looks for the RL wrapper corresponding to a particular rom title */
+/* looks for the RL wrapper corresponding to a particular rom md5,
+then title in that order. returns null if neither match */
 RomSettings* buildRomRLWrapper(const std::string& rom) {
+  std::ifstream romfile(rom);
```

Comment:
This file loading doesn't seem necessary. I would refactor `ALEInterface::loadSettings` to remove the ROM validation check. You should then be able to perform `ALEInterface::loadSettings` in `ALEInterface::loadROM` before calling `buildRomRLWrapper`. Once we've loaded the settings we'll already know the MD5 of the ROM so we don't need to do this extra work. Hopefully that makes sense.

Reply:
Makes sense.

Reply:
So I looked at this further, and I don't think it makes sense anymore. It is a little confusing, but I belive `load_settings` must be called after `romSettings->modifyEnvironmentSettings` which in turn must be called after `buildRomRLWrapper`, making this particular refactor impractical.

I also am not sure of the utility of this change. I mean, loading a ROM is already an extremely expensive operation, I am measuring something like 120ms!!! to load a ROM from the python interface with the following script. I have noticed similar performance from the c++ interace. Surely an additional file load is meaningless in the face of this magnitude of inefficiency.
```
from ale_py import ALEInterface
ale = ALEInterface()
for i in range(100):
    ale.loadROM("wizard_of_wor.bin")
```